hs_core/views/__init__.py

```diff
@@ -69,6 +72,38 @@ def add_file_to_resource(request, shortkey, *args, **kwargs):
     extract_metadata = request.REQUEST.get('extract-metadata', 'No')
     extract_metadata = True if extract_metadata.lower() == 'yes' else False
 
+    irods_fname = request.session.get('irods_add_file_name', '')
+    if irods_fname:
+        res_cls = resource.__class__
+        file_types = res_cls.get_supported_upload_file_types()
```

Comment:
I am following the same pattern of file validation code Pabitra put in JavaScript code without giving it too much thought. After looking into it a bit further, I think the string comparisons are not needed since it does return a tuple. It looks like the for loop in the else block actually does the validation work. Will remove string comparisons and test to make sure it still works. In the meantime, @pkdash could you verify the string comparisons are not needed here for file validation purposes?